protocol/daemons/liquidation/client/client.go

```diff
@@ -65,262 +84,55 @@ func Start(
 	liquidationServiceClient := api.NewLiquidationServiceClient(daemonConn)
 
 	ticker := time.NewTicker(time.Duration(flags.Liquidation.LoopDelayMs) * time.Millisecond)
-	for ; true; <-ticker.C {
-		if err := RunLiquidationDaemonTaskLoop(
-			ctx,
-			flags.Liquidation,
-			subaccountQueryClient,
-			clobQueryClient,
-			liquidationServiceClient,
-		); err != nil {
-			// TODO(DEC-947): Move daemon shutdown to application.
-			logger.Error("Liquidations daemon returned error", "error", err)
-		}
-	}
-
-	return nil
-}
-
-// RunLiquidationDaemonTaskLoop contains the logic to communicate with various gRPC services
-// to find the liquidatable subaccount ids.
-func RunLiquidationDaemonTaskLoop(
-	ctx context.Context,
-	liqFlags flags.LiquidationFlags,
-	subaccountQueryClient satypes.QueryClient,
-	clobQueryClient clobtypes.QueryClient,
-	liquidationServiceClient api.LiquidationServiceClient,
-) error {
-	defer telemetry.ModuleMeasureSince(
-		metrics.LiquidationDaemon,
-		time.Now(),
-		metrics.MainTaskLoop,
-		metrics.Latency,
-	)
+	done := make(chan bool)
 
-	// 1. Fetch all subaccounts from query service.
-	subaccounts, err := GetAllSubaccounts(
+	s := &SubTaskRunnerImpl{}
```

Comment:
nit: things like flags, query clients can just be part of this `SubTaskRunnerImpl` so that we don't need to pass it around 

up to you though

Reply:
Filed a follow-up under daemon lifecycle, since this affects all daemons.